sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/_tests_/e2e/confirmation-of-enrollment.institutions.denyConfirmationOfEnrollment.e2e-spec.ts

```diff
@@ -0,0 +1,227 @@
+  let collegeCLocation: InstitutionLocation;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+
+    const { institution } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation({ institution: collegeC });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+  });
+
+  it("Should decline the COE with provided reason when institution decline a COE.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        institution: collegeC,
+        institutionLocation: collegeCLocation,
+      },
+      { applicationStatus: ApplicationStatus.Enrolment },
+    );
+    const [firstDisbursementSchedule] =
+      application.currentAssessment.disbursementSchedules;
+    const endpoint = `/institutions/location/${collegeCLocation.id}/confirmation-of-enrollment/disbursement-schedule/${firstDisbursementSchedule.id}/deny`;
+    const coeDenyReasonId = 2;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send({ coeDenyReasonId })
+      .auth(
+        await getInstitutionToken(InstitutionTokenTypes.CollegeCUser),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.OK);
+    // Check if the COE is declined and declined with the provided reason.
+    const declinedCOE = await db.disbursementSchedule.findOne({
+      select: { id: true, coeStatus: true, coeDeniedReason: { id: true } },
+      relations: { coeDeniedReason: true },
+      where: { id: firstDisbursementSchedule.id },
+    });
+    expect(declinedCOE.coeStatus).toBe(COEStatus.declined);
+    expect(declinedCOE.coeDeniedReason.id).toBe(coeDenyReasonId);
+  });
+
+  it(
+    "Should decline the COE and create reassessment for impacted application(s) " +
+      "when institution decline a COE and the application that belongs to COE being declined can potentially " +
+      "impact one or more applications.",
+    async () => {
+      // Arrange
+
+      // Create the student to be shared across the applications.
+      const student = await saveFakeStudent(db.dataSource);
+      // Application in the past that must be ignored.
+      const applicationNotImpacted = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          institution: collegeC,
+          institutionLocation: collegeCLocation,
+        },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Completed,
+          currentAssessmentInitialValues: {
+            assessmentWorkflowId: "some fake id",
+            assessmentDate: addDays(-1, new Date()),
```

Comment:
I came to realize later that `addDays(-1, new Date())` can be simplified as `addDays(-1)`.